tests/rest/client/test_relations.py

```diff
@@ -1070,7 +1033,143 @@ def assert_annotations(bundled_aggregations: JsonDict) -> None:
+        self.assertIn(RelationTypes.THREAD, relations)
+        thread_summary = relations[RelationTypes.THREAD]
+        self.assertIn("latest_event", thread_summary)
+
+        # The latest event should not have any bundled aggregations (since it
+        # only has a thread attached.
+        self.assertNotIn("m.relations", thread_summary["latest_event"]["unsigned"])
+
+        # Ensure that requesting the room messages also does not return the sub-thread.
+        channel = self.make_request(
+            "GET",
+            f"/rooms/{self.room}/messages?dir=b",
+            access_token=self.user_token,
+        )
+        self.assertEqual(200, channel.code, channel.json_body)
+        event = self._find_event_in_chunk(channel.json_body["chunk"])
+        relations = event["unsigned"]["m.relations"]
+
+        # The latest event should have bundled aggregations.
+        self.assertIn(RelationTypes.THREAD, relations)
+        thread_summary = relations[RelationTypes.THREAD]
+        self.assertIn("latest_event", thread_summary)
+
+        # The latest event should not have any bundled aggregations (since it
+        # only has a thread attached.
+        self.assertNotIn("m.relations", thread_summary["latest_event"]["unsigned"])
+
+    def test_thread_edit_latest_event(self) -> None:
+        """Test that editing the latest event in a thread works."""
+
+        # Create a thread and edit the last event.
+        channel = self._send_relation(
+            RelationTypes.THREAD,
+            "m.room.message",
+            content={"msgtype": "m.text", "body": "A threaded reply!"},
+        )
+        threaded_event_id = channel.json_body["event_id"]
+
+        new_body = {"msgtype": "m.text", "body": "I've been edited!"}
+        channel = self._send_relation(
+            RelationTypes.REPLACE,
+            "m.room.message",
+            content={"msgtype": "m.text", "body": "foo", "m.new_content": new_body},
+            parent_id=threaded_event_id,
+        )
+        edit_event_id = channel.json_body["event_id"]
+
+        # Fetch the thread root, to get the bundled aggregation for the thread.
+        channel = self.make_request(
+            "GET",
+            f"/rooms/{self.room}/event/{self.parent_id}",
+            access_token=self.user_token,
+        )
+        self.assertEqual(200, channel.code, channel.json_body)
+
+        # We expect that the edit message appears in the thread summary in the
+        # unsigned relations section.
+        relations_dict = channel.json_body["unsigned"].get("m.relations")
+        self.assertIn(RelationTypes.THREAD, relations_dict)
+
+        thread_summary = relations_dict[RelationTypes.THREAD]
+        self.assertIn("latest_event", thread_summary)
+        latest_event_in_thread = thread_summary["latest_event"]
+        self.assertEqual(latest_event_in_thread["content"]["body"], "I've been edited!")
+        # The event should also have edit appear under the bundled aggregations.
```

Comment:
I'm not sure what this comment is trying to say, but it doesn't seem to make sense right now.

Reply:
c48b22056e428bee9458ad473a1fb5acb8bf73ab hopefully clarifies this.